quesma/quesma/config/config.go

```diff
@@ -37,14 +37,15 @@ type QuesmaConfiguration struct {
 	//deprecated
 	ClickHouse RelationalDbConfiguration `koanf:"clickhouse"`
 	//deprecated
-	Hydrolix                   RelationalDbConfiguration     `koanf:"hydrolix"`
-	Elasticsearch              ElasticsearchConfiguration    `koanf:"elasticsearch"`
-	IndexConfig                map[string]IndexConfiguration `koanf:"indexes"`
-	Logging                    LoggingConfiguration          `koanf:"logging"`
-	PublicTcpPort              network.Port                  `koanf:"port"`
-	IngestStatistics           bool                          `koanf:"ingestStatistics"`
-	QuesmaInternalTelemetryUrl *Url                          `koanf:"internalTelemetryUrl"`
-	EnabledOptimizers          OptimizersConfiguration       `koanf:"optimizers"`
+	Hydrolix                      RelationalDbConfiguration             `koanf:"hydrolix"`
+	Elasticsearch                 ElasticsearchConfiguration            `koanf:"elasticsearch"`
+	IndexConfig                   map[string]IndexConfiguration         `koanf:"indexes"`
+	Logging                       LoggingConfiguration                  `koanf:"logging"`
+	PublicTcpPort                 network.Port                          `koanf:"port"`
+	IngestStatistics              bool                                  `koanf:"ingestStatistics"`
+	QuesmaInternalTelemetryUrl    *Url                                  `koanf:"internalTelemetryUrl"`
+	EnabledOptimizers             OptimizersConfiguration               `koanf:"optimizers"`
+	IndexSourceToInternalMappings map[string]IndexMappingsConfiguration `koanf:"indexMappings"`
```

Comment:
This is also out-of-scope of this PR, but I wonder if in the spirit of Quesma wanting to be programmable, maybe the (ingest) transformers could be more dynamic (e.g. their configuration not a static part of configuration, users being able to write their own transformers easily)

Reply:
We can think about that